custom_components/tuya_local/devices/warme_gsh_heater.yaml

```diff
@@ -0,0 +1,38 @@
+name: Warme GSH heater
+products:
+  - id: 8UGflCDtNYJw1dZs
+    name: Warme GSH heater
+primary_entity:
+  entity: climate
+  dps:
+    - id: 1
+      type: boolean
+      name: hvac_mode
+      mapping:
+        - dps_val: true
+          value: "heat"
+          icon: "mdi:radiator"
+        - dps_val: false
+          value: "off"
+          icon: "mdi:radiator-disabled"
+    - id: 2
+      name: temperature
+      type: integer
+      precision: 0
+      unit: "C"
+      range:
+        min: 12
+        max: 50
+    - id: 3
+      name: current_temperature
+      type: integer
+      readonly: true
+      unit: "C"
+    - id: 101
+      name: turn_off_screen
+      type: boolean
+      typeDefaultValue: false
+    - id: 102
+      name: heat_mode
```

Comment:
Could this be preset_mode?  What are the values that it takes?

Reply:
Yes it is preset_mode, I'm updating this section to be....

```
   - id: 102
      name: preset_mode
      type: string
      mapping:
        - dps_val: Comfort
          value: Comfort
        - dps_val: ECO
          value: ECO
        - dps_val: Frost
          value: Frost
      typeDefaultValue: Comfort
```

Reply:
typeDefaultValue has no meaning here,  just remove it.  I asked about the one that was above already because if you meant something by including it, maybe I could tell you how to do what you were trying to do there.

Reply:
Should be resolved now. Also can see additional preset control in HAAS now allowing me to select a preset